meow/user_profile/views.py

```diff
@@ -5,29 +5,146 @@
 from django.core import serializers
 
 from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from meow.utils.decorators import api_login_required
+
 from user_profile.models import User, Theme
 from user_profile.serializers import SafeUserSerializer, ThemeSerializer
 import json
 # Create your views here.
 
+class UserThemes(APIView):
+    def get(self, request, id, format=None):
+        if not request.user.is_authenticated:
+            return HttpResponse('Must be logged in', status=403)
+        user = request.user
+        themes = Theme.objects.filter(name="Daily Bruin")| Theme.objects.filter(name="Dark Bruin") | Theme.objects.filter(author=user)
+        serialized_themes = ThemeSerializer(themes, many=True)
+        themeOrderedDict = serialized_themes.data
+        return JsonResponse(themeOrderedDict, safe=False, status=200)
+    
+    def post(self, request, id, format=None):
+        if not request.user.is_authenticated:
+            return HttpResponse('Must be logged in', status=403)
+        user = request.user
+        themes = Theme.objects.all()
+        req_data = request.data
+        new_name = req_data.get("name", None)
+        new_primary = req_data.get("primary", None)
+        new_secondary = req_data.get("secondary", None)
+        new_primary_font_color = req_data.get("primary_font_color", None)
+        new_secondary_font_color = req_data.get("secondary_font_color", None)
+        new_tertiary = req_data.get("tertiary", None)
+        if new_name == "":
+            return JsonResponse('Theme name cannot be empty', safe=False, status=400)
+        if (themes.filter(name=new_name)):
+            return JsonResponse('Theme name taken, enter a new name', safe=False, status=400)
+        new_theme = Theme.objects.create(primary=new_primary, secondary=new_secondary, primary_font_color=new_primary_font_color, secondary_font_color=new_secondary_font_color, tertiary=new_tertiary, author=user, name=new_name)
+        new_id = Theme.objects.get(name=new_name).pk
+        return JsonResponse(new_id, safe=False, status=200)
+
+    def put(self, request, id, format=None):
+        if not request.user.is_authenticated:
+            return HttpResponse('Must be logged in', status=403)
+        user=request.user
```

Comment:
the problem for the serializers is the theme object from the frontend cant really be serialized straight away because the author is represented by id rather than a user object